src/lug/Graphics/Vulkan/Gui.cpp

```diff
@@ -0,0 +1,876 @@
+#include <lug/Graphics/Vulkan/Gui.hpp>
+
+#include <imgui.h>
+
+#include <lug/Graphics/Vulkan/API/Builder/Buffer.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/CommandBuffer.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/CommandPool.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/DescriptorPool.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/DescriptorSet.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/DescriptorSetLayout.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/DeviceMemory.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/Fence.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/Framebuffer.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/GraphicsPipeline.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/Image.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/ImageView.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/PipelineLayout.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/RenderPass.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/Sampler.hpp>
+#include <lug/Graphics/Vulkan/API/Builder/Semaphore.hpp>
+
+namespace lug {
+namespace Graphics {
+namespace Vulkan {
+
+Gui::Gui(Renderer & renderer, Render::Window & window) : _renderer(renderer), _window(window) {
+}
+
+Gui::~Gui() {
+    for (const auto& frameData: _framesData) {
+        if (frameData.vertexMemoryPtr) {
+            frameData.vertexMemory.unmap();
+        }
+
+        if (frameData.indexMemoryPtr) {
+            frameData.indexMemory.unmap();
+        }
+    }
+}
+
+bool Gui::init(const std::vector<API::ImageView>& imageViews) {
+    ImGuiIO& io = ImGui::GetIO();
+
+    initKeyMapping();
+
+    io.DisplaySize = ImVec2(_window.getWidth(), _window.getHeight());
+    io.DisplayFramebufferScale = ImVec2(1.0f, 1.0f);
+
+    // Get present queue family and retrieve the first queue
```

Comment:
It's just a graphic queue that you want, not the present queue. So rename the variables, etc to match that.